pbl

```diff
@@ -0,0 +1,224 @@
+#! /usr/bin/perl
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# This is a (wrapper) script to update the bootloader config.
+#
+# It checks /etc/sysconfig/bootloader for the bootloader type.
+#
+# If there's no bootloader configured, it does nothing.
+#
+# If the directory /usr/lib/bootloader/$LOADER exists, runs the scripts from
+# that directory.
+#
+# For any other bootloader it runs the old bootloader_entry and
+# update-bootloader scripts (renamed to *.old).
+#
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+use strict;
+use POSIX qw ( strftime uname );
+
+use Data::Dumper;
+$Data::Dumper::Sortkeys = 1;
+$Data::Dumper::Terse = 1;
+$Data::Dumper::Indent = 1;
+
+my $VERSION = "0.0";
+
+my $pbl_dir = "/usr/lib/bootloader";
+
+sub new;
+sub info;
+sub get_bootloader;
+sub run_command;
+
+my $program;
+my $log;
+my $loader;
+my $exit_code = 0;
+my @todo;
+
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+sub new
+{
+  $log->{session_id} = $program . sprintf("-%04d", int rand 10000);
+
+  # log to logfile if we can, else to STDERR
+
+  if(open my $f, ">>/var/log/pbl.log") {
+    my $tmp = select $f;
+    $| = 1;
+    select $tmp;
+    binmode $f, ':utf8';
+    $log->{log_fh} = $f;
+  }
+  elsif(open my $f, ">&STDERR") {
+    binmode $f, ':utf8';
+    $log->{log_fh} = $f;
+    $log->{log_is_stderr} = 1;
+  }
+
+  # find root device & detect if we are chroot-ed
+  my $r = "?";
+  my @r0 = stat "/";
+  my @r1 = stat "/proc/1/root";
+  if(@r0 && @r1) {
+    my $r0 = ($r0[0] >> 8) . ":" . ($r0[0] & 0xff);
+    $r = readlink "/dev/block/$r0";
+    $r =~ s#^..#/dev# if defined $r;
+    $r = $r0 unless defined $r;
+    if($r0[0] != $r1[0] || $r0[1] != $r1[1]) {
+      $r .= " (chroot)";
+    }
+  }
+
+  info(1, "$log->{session_id} = $0, version = $VERSION, root = $r");
+}
+
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+# info(level, message, var, depth)
+#
+# level: 0 .. 3 (debug, info, warning, error)
+# message: log message (single line string)
+# var (optional): either a SCALAR or a REF
+#   - SCALAR (may be multiline) will be logged in a block delimited
+#     by '<<<<<<<<<<<<<<<<' and '>>>>>>>>>>>>>>>>'
+#   - REF will be logged using Data::Dumper
+# depth (optional): maximum depth when logging a REF
+#
+sub info
```

Comment:
log is perl function
I wouldn't really put too much thought into a function name that's called a handfull of times